src/System.Devices.Gpio/System.Devices.Gpio/UnixDriver.cs

```diff
@@ -0,0 +1,182 @@
+using System.Linq;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+
+namespace System.Devices.Gpio
+{
+    public class UnixDriver : GpioDriver
+    {
+        private const string GpioPath = "/sys/class/gpio";
+
+        private ISet<int> _exportedPins;
+
+        public UnixDriver()
+        {
+            _exportedPins = new HashSet<int>();
+        }
+
+        public override void Dispose()
+        {
+            while (_exportedPins.Count > 0)
+            {
+                var pin = _exportedPins.First();
+                UnexportPin(pin);
+            }
+        }
+
+        public override GpioPinMode GetPinMode(int pin)
+        {
+            ExportPin(pin);
+
+            var directionPath = $"{GpioPath}/gpio{pin}/direction";
+            var stringMode = File.ReadAllText(directionPath);
+            var mode = StringModeToPinMode(stringMode);
+            return mode;
+        }
+
+        public override void SetPinMode(int pin, GpioPinMode mode)
+        {
+            ExportPin(pin);
+
+            var directionPath = $"{GpioPath}/gpio{pin}/direction";
+            var stringMode = ModeToStringMode(mode);
+            File.WriteAllText(directionPath, stringMode);
+        }
+
+        public override GpioPinValue Input(int pin)
+        {
+            var valuePath = $"{GpioPath}/gpio{pin}/value";
+            var stringValue = File.ReadAllText(valuePath);
+            var value = StringValueToPinValue(stringValue);
+            return value;
+        }
+
+        public override void Output(int pin, GpioPinValue value)
+        {
+            var valuePath = $"{GpioPath}/gpio{pin}/value";
+            var stringValue = PinValueToStringValue(value);
+            File.WriteAllText(valuePath, stringValue);
+        }
+
+        public override void ClearDetectedEvent(int pin)
+        {
+            throw new NotImplementedException();
+        }
+
+        public override bool EventWasDetected(int pin)
+        {
+            throw new NotImplementedException();
+        }
+
+        public override void SetEventDetection(int pin, GpioEventKind kind, bool enabled)
+        {
+            throw new NotImplementedException();
+        }
+
+        public override bool GetEventDetection(int pin, GpioEventKind kind)
+        {
+            throw new NotImplementedException();
+        }
+
+        public override int PinCount => throw new NotSupportedException();
+
+        public override int ConvertPinNumber(int number, GpioNumberingScheme from, GpioNumberingScheme to)
+        {
+            if (from != GpioNumberingScheme.BCM || to != GpioNumberingScheme.BCM)
+                throw new NotSupportedException("Only BCM numbering scheme is supported");
+
+            return number;
+        }
+
+        #region Private Methods
+
+        private void ExportPin(int pin)
+        {
+            var pinPath = $"{GpioPath}/gpio{pin}";
+
+            if (!Directory.Exists(pinPath))
```

Comment:
I wonder if we should throw if it doesn't exist but it is present in _exportedPins, it seems to indicate some overlapping usage. OTOH we could simply assume if it is in _exportedPins it is good to go (I know something else can change it...). I am not sure about the best path here, we need to think about it.